hs_core/hydroshare/resource.py

```diff
@@ -575,6 +575,12 @@ def create_new_version_resource(ori_res, new_res, user):
     hs_identifier = ori_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
     new_res.metadata.create_element('relation', type='isVersionOf', value=hs_identifier.url)
 
+    if ori_res.resource_type.lower() == "collectionresource":
+        # clone contained_res list of original collection and add to new collection
+        # note that new version collection will not contain "deleted resources"
+        for res_obj in ori_res.resources.all():
```

Comment:
Instead of the for loop, I think it is possible to just do an assignment operation like this:

    new_res.resources = org_res.resources.all()